core/src/test/java/org/infinispan/remoting/transport/InitialClusterSizeTest.java

```diff
@@ -0,0 +1,74 @@
+package org.infinispan.remoting.transport;
+
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.infinispan.commons.CacheException;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.global.GlobalConfigurationBuilder;
+import org.infinispan.manager.EmbeddedCacheManagerStartupException;
+import org.infinispan.test.MultipleCacheManagersTest;
+import org.infinispan.test.fwk.CleanupAfterMethod;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.infinispan.test.fwk.TransportFlags;
+import org.testng.annotations.Test;
+
+@Test(testName = "transport.InitialClusterSizeTest", groups = "functional")
+@CleanupAfterMethod
+public class InitialClusterSizeTest extends MultipleCacheManagersTest {
+   public static final int CLUSTER_SIZE = 4;
+   public static final int CLUSTER_TIMEOUT = 5;
```

Comment:
For timeouts, I find it useful to put the time unit in the constant name, e.g. `CLUSTER_TIMEOUT_SECONDS`.